resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnRMClient.scala

```diff
@@ -42,23 +42,20 @@ private[spark] class YarnRMClient extends Logging {
   /**
    * Registers the application master with the RM.
    *
+   * @param driverHost Host name where driver is running.
+   * @param driverPort Port where driver is listening.
    * @param conf The Yarn configuration.
    * @param sparkConf The Spark configuration.
    * @param uiAddress Address of the SparkUI.
    * @param uiHistoryAddress Address of the application on the History Server.
-   * @param securityMgr The security manager.
-   * @param localResources Map with information about files distributed via YARN's cache.
    */
   def register(
-      driverUrl: String,
-      driverRef: RpcEndpointRef,
+      driverHost: String,
+      driverPort: Int,
```

Comment:
Hi, @vanzin during our internal porting, we found this parameter is misleading.

It should be `amHost` and `amRpcPort` to be more accurate. 
When running on client mode, the value passed here is ApplicationMaster rather than driver.
Do you think it's worth another Jira to resolve this issue?